metric_tank/cassandra.go

```diff
@@ -116,33 +116,38 @@ func searchCassandra(key string, start, end uint32) ([]Iter, error) {
 	query := func(month uint32, q string, p ...interface{}) {
 		wg.Add(1)
 		go func() {
-			//		if len(p) == 3 {
-			//		log.Println("querying cassandra for", q, p[0], TS(p[1]), TS(p[2]))
-			//		} else if len(p) == 2 {
-			//		log.Println("querying cassandra for", q, p[0], TS(p[1]))
-			//		} else {
-			//		log.Println("querying cassandra for", q, p)
-			//		}
 			results <- outcome{month, cSession.Query(q, p...).Iter()}
 			wg.Done()
 		}()
 	}
 
-	start_month := start - (start % month_sec)       // starting row has to be at, or before, requested start
-	end_month := (end - 1) - ((end - 1) % month_sec) // ending row has to be include the last point we might need
+	// unfortunately in the database we only have the t0's of all chunks.
+	// this means we can easily make sure to include the correct last chunk (just query for a t0 < end, the last chunk will contain the last needed data)
+	// but it becomes hard to find which should be the first chunk to include. we can't query for start <= t0 because than we will miss some data,
+	// we effectively need all chunks with a t0 >= start, as well as the last chunk with a t0 <= start which btw, could also be in the previous row.
```

Comment:
The previous chunk can not be in the previous row.  The row_key represents the first Chunks T0. The only time this assertion is not true is when the chunkspan cannot be divided into the 28day row_span without a remainder. which we should just limit.